assets/nix/run.sh

```diff
@@ -5,7 +5,223 @@
 #
 # There are two ways to use this script
 #
-# 1. Via CLI: Run ./run.sh <path to game> [doorstop arguments] -- [game arguments]
+# 1. Via CLI: Run ./run.sh <path to game> [doorstop arguments] [game arguments]
 # 2. Via config: edit the options below and run ./run.sh without any arguments
 
-# TODO: Port script
\ No newline at end of file
+# 0 is false, 1 is true
+
+# LINUX: name of Unity executable
+# MACOS: name of the .app directory
+executable_name=""
+
+# All of the below can be overriden with command line args
+
+# General Config Options
+
+enabled="1"
+
+target_assembly=""
+
+ignore_disable_switch="0"
+
+# Mono Options
+
+search_path_override=""
+
+debug_enable="0"
+
+debug_start_server="0"
+
+debug_address="127.0.0.1:10000"
+
+debug_suspend="0"
+
+# CoreCLR options (IL2CPP)
+
+runtime_path=""
+
+corelib_path=""
+
+
+# Everything past this point is the actual script
+
+# Handle first param being executable name
+if [ -x "$1" ] ; then
+    executable_name="$1"
+    shift
+fi
+
+# Special case: program is launched via Steam
+# In that case rerun the script via their bootstrapper to ensure Steam overlay works
+if [ "$2" = "SteamLaunch" ]; then
+    "$1" "$2" "$3" "$4" "$0" "$5"
+    exit
+fi
```

Comment:
This might break because the `SteamLaunch` path expects that the parameters haven't been shifted yet. I think swapping the order of `if`s should fix this.